I/7802: Update the `Image` feature and API docs for `Insert image via URL` feature. (#7857)
ckeditor/ckeditor5: packages/ckeditor5-image/docs/features/image.md

See the {@link features/image-upload Image upload} guide.

## Insert image via URL

> Another great feature is an ability to insert an image via URL address. The feature is available under the dropdown of the **Image Upload** feature. When you click on the arrow button, you will see a small panel view containing the form for an image source URL. Once you fill it, you will be able to insert the image into the editor's content.

Reviewer: ```suggestion
Another great feature is the ability to insert an image via a URL address. This feature is available under the **Image Upload** feature dropdown. When you click on the arrow button, you will see a small panel view containing the form for an image source URL. Once you fill it, you will be able to insert the image into the editor's content.
```

### Replace image via URL

> Yet, another cool thing that the feature introduces is an ability to replace previously inserted image without loosing its styling and custom attributes (if it has ones). When you click on image and open the `{@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView#dropdownView ImageUpload dropdown}` you will see that the **Insert image via url** form is already filled. This is the value of the selected image. You can easily edit it and update through the form.

Reviewer: ```suggestion
Yet another cool function the feature introduces is the ability to replace a previously inserted image without loosing its styling and custom attributes (if it has ones). If you click on the image and open the `{@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView#dropdownView ImageUpload dropdown}` you will see that the **Insert image via url** form is already filled. This is the URL of the current image. You can easily edit it and update it through the form.
```

<info-box>
<strong>Bonus tip</strong>: You can even edit the URL of an image inserted from the native file system or from — for example — `{@link module:ckfinder/ckfinder~CKFinder}`.
</info-box>

> Here's the short demo:

Reviewer: ```suggestion
Check it out in this short demo:
```

{@snippet features/image-insert-via-url}

### How to add integrations

By default, this feature doesn't need any configuration. You will get a dropdown button, where the main action (icon button) will open a native file dialog and the arrow button will open a dropdown with **Insert image via url** feature.

#### CKFinder

> We also support the integration with {@link module:ckfinder/ckfinder~CKFinder} in that dropdown, but that's something that needs to be configured.

Reviewer: ```suggestion
The integration with {@link module:ckfinder/ckfinder~CKFinder} in that dropdown is also supported, but it needs to be configured.
```

> Here's how you can do this:

Reviewer: ```suggestion
Use this method to do this:
```

```js
ClassicEditor
	.create( document.querySelector( '#editor' ), {
		// ...
		image: {
			// ...
			upload: {
				panel: {
					items: [
						'insertImageViaUrl',
						// Add CKFinder predefined token.
						'openCKFinder'
					]
				}
			}
		},
		ckfinder: {
			uploadUrl: ... // CKFinder upload URL.
		}
	})
```

> As you can see, we added here a new `{@link module:image/imageupload~ImageUploadPanelConfig#items image.upload.panel.items}` configuration, where we defined our desired built-in integrations. Both strings (`insertImageViaUrl` and `openCKFinder`) are special tokens that `{@link module:image/imageupload~ImageUpload}` plugin understands.

Reviewer: ```suggestion
Observe that a new `{@link module:image/imageupload~ImageUploadPanelConfig#items image.upload.panel.items}` configuration defines the desired built-in integrations. Both strings (`insertImageViaUrl` and `openCKFinder`) are special tokens that the `{@link module:image/imageupload~ImageUpload}` plugin understands.
```

Below you can see the demo with built-in CKFinder integration:

{@snippet features/image-insert-via-url-integrations}

#### Custom integrations

If you would want to add your own integration, based on your own plugin ({@link framework/guides/creating-simple-plugin see how to create one}), you have to add that plugin to the editor first and then — like in the configuration from the previous section — you should add your integration component's name (the same you used in the `{@link module:ui/componentfactory~ComponentFactory#add}` when you registered the button) to the `{@link module:image/imageupload~ImageUploadPanelConfig#items}` array.

Here's an example:

```js
// Import your custom plugin.
import PluginX from 'pluginX'

ClassicEditor
	.create( document.querySelector( '#editor' ), {
		// Add your custom plugin to the editor's plugins.
		plugins: [ Image, ImageToolbar, ImageUpload, PluginX ],
		image: {
			upload: {
				panel: {
					items: [
						'insertImageViaUrl',
						'openCKFinder',
						// Use the plugin's registered component.
						'pluginXButton'
					]
				}
			}
		}
	} )
	.then( ... )
	.catch( ... );
```

<info-box>

> Keep in mind that the order of the integrations in configuration equals the order of those views in the `{@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView#dropdownView}`. So in the dropdown panel view you will have: **Insert image via URL** form, next **CKFinder** button and then your custom component for **PluginX**.

Reviewer: ```suggestion
Keep in mind that the order of the integrations in the configuration affects the options order in the `{@link module:image/imageupload/ui/imageuploadpanelview~ImageUploadPanelView#dropdownView}`. Therefore you will have: the **Insert image via URL** form, next the **CKFinder** button and then your custom component for **PluginX** in the dropdown panel view.
```

> You can always change that order if you need.

Reviewer: ```suggestion
You can always change that order if you need by changing the order of the integrations.
```

Reviewer: I hope this is phrased correctly in term of technical names.
